src/xmalloc.c

```diff
@@ -0,0 +1,194 @@
+
+/* xmalloc.c
+ *
+ * Fixed-pool implementation of malloc/free for wolfBoot
+ *
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdlib.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <wolfssl/wolfcrypt/sp.h>
+#include "target.h"
+
+
+struct xmalloc_slot {
+    uint8_t *addr;
+    uint32_t size;
+    uint32_t in_use;
+};
+
+#define SP_DIGIT_SIZE (4)
```

Comment:
Is this true for 64 bit targets?

Reply:
Good point. changed to `sizeof(sp_digit)`